src/main/java/org/cryptomator/cryptofs/CryptoFileSystemModule.java

```diff
@@ -37,6 +39,20 @@ public Cryptor provideCryptor(CryptorProvider cryptorProvider, @PathToVault Path
 		}
 	}
 
+	private String generateFileIdIfPossible(byte[] fileBytes) {
+		try {
+			MessageDigest md = MessageDigest.getInstance("SHA-256");
+			byte[] truncatedDigest = md.digest(fileBytes);
+			StringBuilder sb = new StringBuilder();
+			for (int i = 0; i < 4; i++) { // we only need the first 4 bytes, since each is translated into two characters
+				sb.append(String.format("%02x", truncatedDigest[i]));
+			}
+			return sb.toString().substring(0, 7);
```

Comment:
Easier using Guava's [BaseEncoding](https://guava.dev/releases/27.1-jre/api/docs/com/google/common/io/BaseEncoding.html#encode-byte:A-int-int-):

```suggestion
			return BaseEncoding.base16().lowerCase().encode(digest, 0, 4);
```

Reply:
And let's just use 8 chars (i.e. 4 bytes) instead of 7. :wink: